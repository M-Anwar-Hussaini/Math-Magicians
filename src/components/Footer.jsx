import { useEffect, useState } from 'react';

function Footer() {
  const [time, setTime] = useState(new Date().toLocaleTimeString());
  useEffect(() => {
    setInterval(() => setTime(new Date().toLocaleTimeString()), 1000);
  }, [time]);
  return (
    <footer className="p3 bg-success-subtle mt-auto d-flex justify-content-between container align-items-center rounded">
      <div>{time}</div>
      <h3>Developed by: Mohammad Anwar Hussaini</h3>
    </footer>
  );
}

export default Footer;
