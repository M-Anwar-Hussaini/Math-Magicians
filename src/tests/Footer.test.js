import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '../components/Footer';

describe('Test the funtionalities of footer', () => {
  const footerEl = render(<Footer />);

  // Create snapshot for later matching
  test('Match the snapshot', () => {
    expect(footerEl).toMatchSnapshot();
  });

  // Test whether this component contains a footer element
  test('Footer components contains a footer html tag', () => {
    render(<Footer />);
    const footerTag = screen.getByRole('contentinfo');
    expect(footerTag).toBeInTheDocument();
  });

  // Test whether this component contains a heading element
  test('Footer components contains a heading html tag', () => {
    render(<Footer />);
    const headingTag = screen.getByRole('heading');
    expect(headingTag).toBeInTheDocument();
  });
});
