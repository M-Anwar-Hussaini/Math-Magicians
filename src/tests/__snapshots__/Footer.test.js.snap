// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the funtionalities of footer Match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="p3 bg-success-subtle mt-auto d-flex justify-content-between container align-items-center rounded"
      >
        <div>
          12:45:13 AM
        </div>
        <h3>
          Developed by: Mohammad Anwar Hussaini
        </h3>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="p3 bg-success-subtle mt-auto d-flex justify-content-between container align-items-center rounded"
    >
      <div>
        12:45:13 AM
      </div>
      <h3>
        Developed by: Mohammad Anwar Hussaini
      </h3>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
